### PR TITLE
Validations/alvaro

### PR DESCRIPTION
**Pull request by junior0123**

### CODE REVIEW

**Review by rodri-98 — COMMENTED** [review]
Please add documentation for the new classes that were added

File: app/src/main/java/ju/video/player/model/request/AudioRequest.java [hunk 1]
```diff
@@ -45,30 +47,36 @@ public AudioRequest () {
...
-        HttpPost httppost = new HttpPost(HTTP_POST);
-        FileBody bin = new FileBody(new File(path));
-        StringBody bitrate = new StringBody("");
-        StringBody channels = new StringBody("");
-        StringBody samplingFrequency  = new StringBody("");
-        StringBody outFormat = new StringBody(newFormat);
+        try {
+            validateFormat();
+            validatePath();
+            HttpClient httpclient = new DefaultHttpClient();
```
Comment (MariaAT18) [inline app/src/main/java/ju/video/player/model/request/AudioRequest.java]: Keep to variables lowerCamelCase => httpClient

File: app/src/main/java/ju/video/player/model/request/AudioRequest.java [hunk 2]
```diff
@@ -45,30 +47,36 @@ public AudioRequest () {
...
+            HttpPost httppost = new HttpPost(HTTP_POST);
```
Comment (MariaAT18) [inline app/src/main/java/ju/video/player/model/request/AudioRequest.java]: Keep to variables lowerCamelCase =>httpPost

File: app/src/main/java/ju/video/player/model/request/AudioRequest.java [hunk 3]
```diff
@@ -45,30 +47,36 @@ public AudioRequest () {
...
+            FileBody bin = new FileBody(new File(path));
```
Comment (MariaAT18) [inline app/src/main/java/ju/video/player/model/request/AudioRequest.java]: "bin", explain what does mean this variable in you declaration. We need to declare that the variables are significant.

File: app/src/main/java/ju/video/player/model/request/AudioRequest.java [hunk 4]
```diff
@@ -45,30 +47,36 @@ public AudioRequest () {
...
+            StringBody bitrate = new StringBody("");
```
Comment (MariaAT18) [inline app/src/main/java/ju/video/player/model/request/AudioRequest.java:62]: Keep to variables lowerCamelCase

File: app/src/main/java/ju/video/player/model/request/AudioRequest.java [hunk 5]
```diff
@@ -82,27 +90,32 @@ public String getOutputPath() {
...
-            byte[] bytes = new byte[2048];
-            int length;
-            OutputStream out = new FileOutputStream("Download\\" + separateFormat(getName(path)) + "." + newFormat);
-            outputPath += System.getProperty("user.dir") + "\\Download\\" + separateFormat(getName(path)) + "." + newFormat;
-            while ((length = in.read(bytes)) != -1) {
-                out.write(bytes, 0, length);
+    private void sendGet() throws AudioRequestException{
+        try {
+            URL url = new URL(HTTP_GET + getName(path));
+            HttpURLConnection conn = (HttpURLConnection) url.openConnection();
```
Comment (MariaAT18) [inline app/src/main/java/ju/video/player/model/request/AudioRequest.java]: conn can be improved as variable.

File: app/src/main/java/ju/video/player/model/request/AudioRequest.java [hunk 6]
```diff
@@ -82,27 +90,32 @@ public String getOutputPath() {
...
+            conn.setRequestMethod("GET");
+            conn.connect();
+            int responseCode = conn.getResponseCode();
+            if (responseCode != 200) {
+                throw new RuntimeException("An error has occurred: " + responseCode);
+            } else {
+                InputStream in = conn.getInputStream();
```
Comment (MariaAT18) [inline app/src/main/java/ju/video/player/model/request/AudioRequest.java]: Inprove "in" as declaration variable

File: app/src/main/java/ju/video/player/model/request/AudioRequest.java [hunk 7]
```diff
@@ -82,27 +90,32 @@ public String getOutputPath() {
...
+                byte[] bytes = new byte[2048];
+                int length;
+                OutputStream out = new FileOutputStream("Download\\" + separateFormat(getName(path)) + "." + newFormat);
```
Comment (MariaAT18) [inline app/src/main/java/ju/video/player/model/request/AudioRequest.java]: Improve "out" as a declaration variable

File: app/src/main/java/ju/video/player/model/request/AudioRequest.java [hunk 8]
```diff
@@ -82,27 +90,32 @@ public String getOutputPath() {
...
+                outputPath += System.getProperty("user.dir") + "\\Download\\" + separateFormat(getName(path)) + "." + newFormat;
```
Comment (MariaAT18) [inline app/src/main/java/ju/video/player/model/request/AudioRequest.java:109]: Keep lowerCamelCase to variables.

File: app/src/main/java/ju/video/player/model/request/VideoRequest.java [hunk 9]
```diff
@@ -48,45 +52,51 @@ public class VideoRequest implements  Request{
...
-        reqEntity.addPart("size", size);
-        reqEntity.addPart("cropVideo", cropVideo);
-        httppost.setEntity(reqEntity);
-        HttpResponse response = httpclient.execute(httppost);
-        HttpEntity resEntity = response.getEntity();
-        sendGet();
+        try {
+            validateFormat();
+            validatePath();
+            HttpClient httpclient = new DefaultHttpClient();
```
Comment (MariaAT18) [inline app/src/main/java/ju/video/player/model/request/VideoRequest.java]: Keep lowerCamelCase to variables declaration

File: app/src/main/java/ju/video/player/model/request/VideoRequest.java [hunk 10]
```diff
@@ -48,45 +52,51 @@ public class VideoRequest implements  Request{
...
+            HttpPost httppost = new HttpPost(HTTP_POST);
```
Comment (MariaAT18) [inline app/src/main/java/ju/video/player/model/request/VideoRequest.java]: Keep lowerCamelCase to variables declaration

File: app/src/main/java/ju/video/player/model/request/VideoRequest.java [hunk 11]
```diff
@@ -100,29 +110,32 @@ public String getOutputPath() {
...
-            throw new RuntimeException("An error has occurred: " + responseCode);
-        } else {
-            InputStream in = conn.getInputStream();
-            byte[] bytes = new byte[2048];
-            int length;
-            OutputStream out = new FileOutputStream("Download\\" + separateFormat(getName(path)) + "." + newFormat);
-            outputPath += System.getProperty("user.dir") + "\\Download\\" + separateFormat(getName(path)) + "." + newFormat;
-            while ((length = in.read(bytes)) != -1) {
-                out.write(bytes, 0, length);
+    private void sendGet() throws VideoRequestException {
```
Comment (MariaAT18) [inline app/src/main/java/ju/video/player/model/request/VideoRequest.java]: To this method, the same advice that before.

**Review by MariaAT18 — CHANGES_REQUESTED** [review]

File: app/src/main/java/ju/video/player/model/request/AudioRequest.java (hunk 4 above)
Comment (junior0123) [reply app/src/main/java/ju/video/player/model/request/AudioRequest.java:62]: Done

File: app/src/main/java/ju/video/player/model/request/AudioRequest.java (hunk 8 above)
Comment (junior0123) [reply app/src/main/java/ju/video/player/model/request/AudioRequest.java:109]: Done

Comment (junior0123) [reply app/src/main/java/ju/video/player/model/request/AudioRequest.java:109]: Done

File: app/src/main/java/ju/video/player/model/request/AudioRequest.java (hunk 4 above)
Comment (junior0123) [reply app/src/main/java/ju/video/player/model/request/AudioRequest.java:62]: Done

**Review by rodri-98 — APPROVED** [review]
Good job

**Pull request merged**